### PR TITLE
[GPU] Update debug config verbose option to show the name of tensors allocated

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - Update debug config verbose option to show the name of tensors allocated

### Tickets:
 - 63233

### CODE REVIEW

**Review by lznamens — APPROVED** [review]
This PR looks good to me.

File: inference-engine/thirdparty/clDNN/src/primitive_inst.cpp [hunk 1]
```diff
@@ -174,21 +175,30 @@ memory::ptr primitive_inst::allocate_output() {
...
+            GPU_DEBUG_COUT << "[" << _node.id() << ": output]" << std::endl;
+        }
         return _network.get_memory_from_pool(layout,
                                              _node.id(),
                                              _node.get_memory_dependencies(),
                                              alloc_type,
                                              false);
     } else if (_network.is_internal() && _node.is_output() && _node.is_type<generic_layer>() &&
                engine.supports_allocation(allocation_type::usm_device)) {
+        GPU_DEBUG_IF(debug_config->verbose >= 1) {
```
Comment (vladimir-paramuzov) [inline inference-engine/thirdparty/clDNN/src/primitive_inst.cpp:190]: What do you think about increase of verbose level for such prints to 2?

File: inference-engine/thirdparty/clDNN/src/primitive_inst.cpp [hunk 2]
```diff
@@ -174,21 +175,30 @@ memory::ptr primitive_inst::allocate_output() {
...
+            GPU_DEBUG_COUT << "[" << _node.id() << ": output]" << std::endl;
+        }
         return engine.allocate_memory(layout, allocation_type::usm_device, false);
     } else if (_network.is_internal() && !_node.is_output() && _node.is_type<input_layout>()) {
         // Skip memory reset for input_layout primitives, since data will be copied from cldnn::data primitive
         // or just reuse primitive's memory
         return engine.allocate_memory(layout, alloc_type, false);
```
Comment (vladimir-paramuzov) [inline inference-engine/thirdparty/clDNN/src/primitive_inst.cpp:197]: Why this branch is not covered?

File: inference-engine/thirdparty/clDNN/src/primitive_inst.cpp (hunk 1 above)
Comment (andrew-k-park) [reply inference-engine/thirdparty/clDNN/src/primitive_inst.cpp]: I agree. done.

File: inference-engine/thirdparty/clDNN/src/primitive_inst.cpp (hunk 2 above)
Comment (andrew-k-park) [reply inference-engine/thirdparty/clDNN/src/primitive_inst.cpp:200]: done

**Comment by andrew-k-park** [comment]
@vladimir-paramuzov All comments are applied and CI checks are passed. Could you review this PR again?

File: inference-engine/src/cldnn_engine/ops/non_max_suppression.cpp [hunk 3]
```diff
@@ -74,6 +76,9 @@ void CreateNonMaxSuppressionIEInternalOp(Program& p, const std::shared_ptr<ngrap
                 DefaultFormatForDims(op->get_output_shape(2).size()),
                 CldnnTensorFromIEDims(op->get_output_shape(2)));
 
+            GPU_DEBUG_IF(debug_config->verbose >= 2) {
+                GPU_DEBUG_COUT << "[" << layer_type_name_ID(op) << ": mutable data]" << std::endl;
+            }
```
Comment (sshlyapn) [inline inference-engine/src/cldnn_engine/ops/non_max_suppression.cpp:81]: Why did you ignore some other ops with mutable data/internal memory allocations, such as proposal, topk, greedy decoder?

Comment (andrew-k-park) [reply inference-engine/src/cldnn_engine/ops/non_max_suppression.cpp:81]: I just missed those ops. updated.

**Review by vladimir-paramuzov — APPROVED** [review]

**Review by sshlyapn — APPROVED** [review]

**Comment by andrew-k-park** [comment]
@vladimir-paramuzov Could you also enable auto-merge for this PR?

**Pull request merged**